### PR TITLE
FIX: Add env vars for integration tests

### PR DESCRIPTION
**Pull request by ucokzeko**

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -26,6 +26,6 @@ in rec {
   # Will be run in a container with all Detox services running
   integrationTest = ''
     cd ${test}/lib/node_modules/mqtt-play
-    INTEGRATION_TESTING=1 ${pkgs.nodejs}/bin/npm run integrationTests
+    INTEGRATION_TESTING=1 MOSQUITTO_ADDRESS=http://localhost:1883 PLAY_COMMAND=${pkgs.sox}/bin/play ${pkgs.nodejs}/bin/npm run integrationTests
```
Comment (dwillie) [inline release.nix]: Integration tests occur in a containerised environment set up by our continuous integration service and mosquitto might not be running at that address when your tests run, it should be running at the address as defined in defaultEnvironment of detox-declare/services/default.nix (which may change independent of this service)

@rimmington My understanding is that variable should be set for all services by these integration tests are crashing as it is undefined for them, what are we doing wrong here?

Comment (rimmington) [inline release.nix]: https://github.com/dstil/mqtt-play/commit/58efb530494fa105d3870027e36d713654b7d2d9 added a `consts` import, which crashes if some env vars not necessary for integration tests are not defined.

Comment (rimmington) [inline release.nix]: Note that the env vars are defined for the services, but the integration tests are not a service.

Comment (dwillie) [inline release.nix]: The env vars are necessary to address the detox api and the mosquitto queue though, which the integration tests do. Is there some other way we could approach testing these?

Comment (dwillie) [inline release.nix]: @rimmington

Comment (rimmington) [inline release.nix]: Point. I suppose you could add `defaultEnvironment` to all the integration tests by extracting them into their own file and modifying `detox-declare/release.nix`. You could do the same for service-specific vars if you needed to. The only disadvantage is the proliferation of .nix files.

**Comment by dwillie** [comment]
@ucokzeko This shouldn't be necessary any more as I've updated detox declare to include the environment variables in integration tests now :)

**Comment by ucokzeko** [comment]
@dwillie Cool
I'll close this PR then

**Pull request closed without merging**